Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java

```diff
@@ -142,7 +148,74 @@ private void updateTagsView (SheetAdapter.SheetViewHolder holder, SheetModel mod
             String pickupChargesText = formattedPickupChargesText;
             String searchRequestId = model.getSearchRequestId();
             boolean showVariant =  !model.getRequestedVehicleVariant().equals(NO_VARIANT) && model.isDowngradeEnabled() && RideRequestUtils.handleVariant(holder, model, this);
-            if (model.getCustomerTip() > 0 || model.getDisabilityTag() || searchRequestId.equals(DUMMY_FROM_LOCATION) || model.isGotoTag() || showVariant || showSpecialLocationTag) {
+            if(model.getRideProductType().equals(RENTAL)){
```

Comment:
Create A different function `updateRentalData` in RideReqUtils and use it across both OverlaySheetService and RideReqActivity

You can Refer to `updateTierAndAC`